app/services/definitions.py

```diff
@@ -0,0 +1,543 @@
+    """
+    bound_logger = logger.bind(body=schema)
+
+    deps = ServiceDependencies(bound_logger, aries_controller)
+    publisher = SchemaPublisher(deps)
+    registrar = SchemaRegistrar(deps)
+
+    try:
+        result = await publisher.publish_schema(schema_request)
+    except CloudApiException as e:
+        if "already exist" in e.detail and e.status_code == 400:
+            result = await publisher.handle_existing_schema(schema)
+        else:
+            bound_logger.warning(
+                f"An unhandled Exception was caught while publishing schema: {e.detail}"
+            )
+            raise CloudApiException("Error while creating schema.") from e
+
+    if result.sent and result.sent.schema_id:
+        await registrar.register_schema(result.sent.schema_id)
+    else:
+        bound_logger.error("No SchemaSendResult in `publish_schema` response.")
+        raise CloudApiException(
+            "An unexpected error occurred: could not publish schema."
+        )
+
+    result = credential_schema_from_acapy(result.sent.var_schema)
+    bound_logger.info("Successfully published and registered schema.")
+    return result
+
+
+async def get_schemas_tenant(
+    aries_controller: AcaPyClient,
+    schema_id: Optional[str],
+    schema_issuer_did: Optional[str],
+    schema_name: Optional[str],
+    schema_version: Optional[str],
+) -> List[CredentialSchema]:
+    """
+    Allows tenants to get all schemas created
+    """
+    bound_logger = logger.bind(
+        body={
+            "schema_id": schema_id,
+            "schema_issuer_did": schema_issuer_did,
+            "schema_name": schema_name,
+            "schema_version": schema_version,
+        }
+    )
+    bound_logger.debug("Fetching schemas from trust registry")
+
+    if not schema_id:  # client is not filtering by schema_id, fetch all
+        trust_registry_schemas = await get_trust_registry_schemas()
+    else:  # fetch specific id
+        trust_registry_schemas = [await get_trust_registry_schema_by_id(schema_id)]
+
+    schema_ids = [schema.id for schema in trust_registry_schemas]
+
+    bound_logger.debug("Getting schemas associated with fetched ids")
+    schemas = await schema_futures(schema_ids, aries_controller)
+
+    if schema_issuer_did:
+        schemas = [
+            schema for schema in schemas if schema.id.split(":")[0] == schema_issuer_did
+        ]
+    if schema_name:
+        schemas = [schema for schema in schemas if schema.name == schema_name]
+    if schema_version:
+        schemas = [schema for schema in schemas if schema.version == schema_version]
+
+    return schemas
+
+
+async def get_schemas_governance(
+    aries_controller: AcaPyClient,
+    schema_id: Optional[str],
+    schema_issuer_did: Optional[str],
+    schema_name: Optional[str],
+    schema_version: Optional[str],
+) -> List[CredentialSchema]:
+    """
+    Governance agents gets all schemas created by itself
+    """
```

Comment:
There should be some assertion at the start of this method which asserts that the controller is a governance controller. Otherwise this method can be called using any controller